test/units/modules/network/f5/test_bigip_gtm_dns_listener.py

```diff
@@ -0,0 +1,299 @@
+        assert results['irules'][0] == '/Common/rule1' and results['irules'][1] == '/Common/rule2'
+
+    def test_delete(self, *args):
+        set_module_args(dict(
+            name='test-dns-listener',
+            address='192.10.0.2',
+            state='absent',
+            provider=dict(
+                server='localhost',
+                password='password',
+                user='admin'
+            )
+        ))
+
+        module = AnsibleModule(
+            argument_spec=self.spec.argument_spec,
+            supports_check_mode=self.spec.supports_check_mode
+        )
+
+        current = ApiParameters(
+            dict(
+                agent_status_traps='disabled'
+            )
+        )
+
+        # Override methods to force specific logic in the module to happen
+        mm = ModuleManager(module=module)
+        mm.exists = Mock(return_value=False)
+        results = mm.exec_module()
+
+        assert results['changed'] is False
+
+    def test_enable_listener(self, *args):
+        set_module_args(dict(
+            name='test-dns-listener',
+            state='enabled',
+            address='192.10.0.2',
+            provider=dict(
+                server='localhost',
+                password='password',
+                user='admin'
+            )
+        ))
+
+        module = AnsibleModule(
+            argument_spec=self.spec.argument_spec,
+            supports_check_mode=self.spec.supports_check_mode
+        )
+
+        current = ApiParameters(
+            dict(
+                agent_status_traps='disabled'
+            )
+        )
+
+        # Override methods to force specific logic in the module to happen
+        mm = ModuleManager(module=module)
+        mm.exists = Mock(return_value=True)
+        mm.update_on_device = Mock(return_value=True)
+        mm.read_current_from_device = Mock(return_value=current)
+        results = mm.exec_module()
+
+        assert results['changed'] is True
+
+    def test_disable_listener(self, *args):
+        set_module_args(dict(
+            name='test-dns-listener',
+            state='disabled',
+            address='192.10.0.2',
+            provider=dict(
+                server='localhost',
+                password='password',
+                user='admin'
+            )
+        ))
+
+        module = AnsibleModule(
+            argument_spec=self.spec.argument_spec,
+            supports_check_mode=self.spec.supports_check_mode
+        )
+
+        current = ApiParameters(
+            dict(
+                agent_status_traps='disabled'
+            )
+        )
+
+        # Override methods to force specific logic in the module to happen
+        mm = ModuleManager(module=module)
+        mm.exists = Mock(return_value=True)
+        mm.update_on_device = Mock(return_value=True)
+        mm.read_current_from_device = Mock(return_value=current)
+        results = mm.exec_module()
+
+        assert results['changed'] is True
```

Comment:
maybe test the 'all' vlan scenario here

Reply:
I have added unit tests to validate enabled_vlans:all and disabled_vlans:all cases
This comment is addressed now